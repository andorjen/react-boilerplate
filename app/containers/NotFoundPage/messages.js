/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage container.
 */
import { defineMessages } from 'react-intl';

export const scope = 'app.containers.NotFoundPage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: "OOPS, WE CAN'T FIND WHAT YOU ARE LOOKING FOR :(",
  },
});
