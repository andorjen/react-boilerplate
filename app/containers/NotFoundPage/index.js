/**
 * NotFoundPage
 *
 * This is the page we show when the user visits a url that doesn't have a route
 *
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';
import styled from 'styled-components';
import NavBar from '../../components/NavBar';
import messages from './messages';
const Header = styled.div`
  text-align: center;
  height: 6em;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgb(250, 234, 112);
  margin: 1em 1.6em 0 1.6em;
  border-radius: 0.4em;
`;

export default function NotFound() {
  return (
    <div>
      <NavBar />
      <Header>
        <h1 style={{ textAlign: 'center' }}>
          <FormattedMessage {...messages.header} />
        </h1>
      </Header>
    </div>
  );
}
