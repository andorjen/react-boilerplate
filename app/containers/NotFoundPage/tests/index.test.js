import React from 'react';
import { render } from 'react-testing-library';
// import { IntlProvider } from 'react-intl';

import 'jest-dom/extend-expect';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import history from 'utils/history';
import configureStore from '../../../configureStore';
import NotFoundPage from '../index';

// Create redux store with history
const initialState = {};
const store = configureStore(initialState, history);

describe('<NotFoundPage />', () => {
  it('should render and match the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <NotFoundPage />
        </MemoryRouter>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
