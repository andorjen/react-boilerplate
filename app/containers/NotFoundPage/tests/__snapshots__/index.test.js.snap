// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> should render and match the snapshot 1`] = `
<div>
  <div>
    <nav
      class="NavBar__Nav-ar0vx-0 haeefk"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
        style="background-color: rgba(140, 71, 245, 0.4); font-weight: bold;"
      >
        Home
      </a>
      <a
        href="/add"
      >
        Add A Card
      </a>
    </nav>
    <div
      class="NotFoundPage__Header-sc-8x6wqx-0 bUbrSb"
    >
      <h1
        style="text-align: center;"
      >
        <span>
          OOPS, WE CAN'T FIND WHAT YOU ARE LOOKING FOR
        </span>
      </h1>
    </div>
  </div>
</div>
`;
